src/proteus/dswx_hls.py

```diff
@@ -3224,11 +3233,17 @@ def _warp(input_file, geotransform, projection,
                            tile_max_x_utm + margin_m,
                            tile_srs, file_srs)
 
-    # if it doesn't cross the antimeridian, reproject
-    # input file using gdalwarp
-    if not (file_srs.IsGeographic() and tile_min_x < 180 and
-            tile_max_x >= 180):
-
+    # The dateline crossing by the tile is tested by the condition:
+    #     `tile_min_x < 180 and tile_max_x >= 180`
+    # The ancillary input only requires reprojection if it is in
+    # geographic coordinates and if its longitude domain is represented
+    # within the [-180, +180] range, verified by the test `min_x < -170`,
+    # rather than the [0, +360] interval.
+    if not (file_srs.IsGeographic() and file_min_x < -170 and
+            tile_min_x < 180 and tile_max_x >= 180):
+
+        # if it doesn't cross the antimeridian, reproject
+        # input file using gdalwarp
```

Comment:
Perhaps this could be a bit more straightforward?

```suggestion
    # Flag for if the geotransform (of the e.g. HLS tile) crosses the dateline.
    # The dateline crossing by the tile is tested by the condition:
    #     `tile_min_x < 180 and tile_max_x >= 180`
    flag_tile_crosses_dateline = tile_min_x < 180 and tile_max_x >= 180

    # Flag for if the ancillary input file crosses the dateline.
    # The ancillary input only requires reprojection if it is in
    # geographic coordinates and if its longitude domain is represented
    # within the [-180, +180] range, verified by the test `min_x < -170`,
    # rather than the [0, +360] interval.
    flag_file_crosses_dateline = file_srs.IsGeographic() and file_min_x < -170

    # If neither crosses the antimeridian, reproject input file using gdalwarp
    if not (flag_tile_crosses_dateline and flag_file_crosses_dateline):
```